### PR TITLE
Fix and simplify setupRepos

### PR DESCRIPTION
**Pull request by montehurd**
- Directly use shell commands to setup repos instead tunneling though Node complexities
- Perform setups sequentially so it's easier see what's happening and exactly where any failures are occurring
- Explicitly exit with error immediacy upon critical step failure
- Print more actionable error messages
- Print composer console output only if something goes wrong (its volume of output makes it hard to follow what's happening otherwise)
- Only run composer update if a repo has a composer.json file (Codex doesn't - was causing a failure)

### CODE REVIEW

**Comment by montehurd** [comment]
I mentioned this to Bernard:

> I think the initial motivation in using Node for setupRepos bits was to achieve some parallelization of the repo-fetching process, but this came at the cost of making it harder to debug when something goes wrong

I think my Node-less approach in this PR is going to be easier to maintain

**Comment by bernardwang** [comment]
Fixes my setup repo issue!
